### PR TITLE
Update templates for documentation pages

### PR DESCRIPTION
**Pull request by Girgias**
Doc gen is generating outdated markup already, especially for classes.

### CODE REVIEW

File: skeletons/classname.xml [hunk 1]
```diff
@@ -1,105 +1,132 @@
...
+   <classsynopsis class="class">
+    <ooclass>
+     <classname>ClassName</classname>
+    </ooclass>
+
+    <!-- If the class extends another one, use this -->
+    <ooclass>
+     <modifier>extends</modifier>
+     <classname>ParentClass</classname>
+    </ooclass>
```
Comment (cmb69) [inline skeletons/classname.xml:30]: Not 100% sure that uncommenting is a good idea, although that's a hint for the author to remove unnecessary sections.

File: skeletons/classname.xml [hunk 2]
```diff
@@ -1,105 +1,132 @@
...
      <type>string</type>
-     <varname linkend="classname.props.name">name</varname>
+     <varname linkend="classname.props.someproperty">SomeProperty</varname>
+     <initializer>""</initializer>
     </fieldsynopsis>
--->
- 
-<!-- If properties are documented "as a method", use this
-    <xi:include xpointer="xmlns(db=http://docbook.org/ns/docbook) xpointer(id('class.classname')/db:refentry/db:refsect1[@role='description']/descendant::db:fieldsynopsis)" />
--->
```
Comment (cmb69) [inline skeletons/classname.xml:60]: Not sure about this removal. Properties probably have never been documented in separate files, though.

File: skeletons/classname.xml [hunk 3]
```diff
@@ -1,105 +1,132 @@
...
+     <xi:fallback/>
+    </xi:include>
+
+    <!-- Edit the classname below -->
+    <!-- If has a constructor -->
     <classsynopsisinfo role="comment">&Methods;</classsynopsisinfo>
-    <xi:include xpointer="xmlns(db=http://docbook.org/ns/docbook) xpointer(id('class.classname')/db:refentry/db:refsect1[@role='description']/descendant::db:constructorsynopsis[not(@role='procedural')])">
-     <xi:fallback />
+    <xi:include xpointer="xmlns(db=http://docbook.org/ns/docbook) xpointer(id('class.classname')/db:refentry/db:refsect1[@role='description']/descendant::db:constructorsynopsis[@role='ClassName'])">
+     <xi:fallback/>
```
Comment (cmb69) [inline skeletons/classname.xml:71]: Likely a good idea, but might be a small footgun for authors not updating `ClassName` here.

File: skeletons/function.xml [hunk 4]
```diff
@@ -1,65 +1,67 @@
...
+  <refpurpose>The function_name purpose</refpurpose>
  </refnamediv>
 
- <refsect1 role="description"><!-- {{{ -->
+ <refsect1 role="description">
   &reftitle.description;
   <methodsynopsis role="procedural">
    <!-- Example: All functions have this -->
-   <type>thereturned type</type><methodname>func_name</methodname>
+   <type>thereturned type</type><methodname>function_name</methodname>
```
Comment (cmb69) [inline skeletons/function.xml:12]:
```suggestion
   <type>the returned type</type><methodname>function_name</methodname>
```

File: skeletons/function.xml [hunk 5]
```diff
@@ -108,31 +110,31 @@ Use the PEAR Coding standards
 Use a screen like we did below
 ]]>
    </screen>
-  </example><!-- }}} -->
- </refsect1><!-- }}} -->
+  </example>
+ </refsect1>
```
Comment (cmb69) [inline skeletons/function.xml:114]: Thanks for removing the folding markers. :)

File: skeletons/method.xml [hunk 6]
```diff
@@ -20,11 +20,11 @@ REFNAME:  Procedural refname goes after the OOP refname.
  <refsect1 role="description">
   &reftitle.description;
 <!-- IF OOP & PROCEDURAL.
-  <para>&style.oop;</para>
+  <simpara>&style.oop;</para>
```
Comment (cmb69) [inline skeletons/method.xml:23]:
```suggestion
  <simpara>&style.oop;</simpara>
```

**Review by cmb69 — COMMENTED** [review]
Thank you! Looks generally good to me.

File: skeletons/function.xml (hunk 5 above)
Comment (jimwins) [reply skeletons/function.xml:114]: Should we add a note to the style guide that these are discouraged?

File: skeletons/classname.xml (hunk 1 above)
Comment (jimwins) [reply skeletons/classname.xml:30]: In general, I think we don't want commented-out boilerplate in skeletons, and there should be dummy text in anything that is expected to be filled out. We should add a QA script that trips on dummy text left in new files.

**Review by jimwins — APPROVED** [review]
LGTM once @cmb69's notes are addressed.

File: skeletons/function.xml (hunk 5 above)
Comment (Girgias) [reply skeletons/function.xml:114]: Probably yes

File: skeletons/classname.xml (hunk 2 above)
Comment (Girgias) [reply skeletons/classname.xml:60]: mysqli does this AFAIK

**Review by MauricioFauth — APPROVED** [review]

**Pull request merged**